### PR TITLE
add EventGridTrigger support

### PR DESCRIPTION
**Pull request by christopheranderson**
fixes https://github.com/Azure/azure-functions-java-worker/issues/58

(Still need to test this, so do not merge it yet :))

### CODE REVIEW

**Comment by JunyiYi** [comment]
It would be better to also upgrade the following versions to `1.0.0-beta-3`:
* https://github.com/Azure/azure-functions-java-worker/blob/dev/azure-functions-java-core/pom.xml#L5
* https://github.com/Azure/azure-functions-java-worker/blob/dev/azure-functions-java-worker/pom.xml#L47

And update the version below to `1.0.0-beta-4` as well:
* https://github.com/Azure/azure-functions-java-worker/blob/dev/azure-functions-java-worker/pom.xml#L5

**Comment by kevinhillinger** [comment]
Looking into supporting the binding in the maven plugin, I noticed that there seems to be an implicit contract being created where `name()` and `dataType()` is the minimum definition for a binding.

see [BaseBinding](https://github.com/Microsoft/azure-maven-plugins/blob/master/azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/bindings/BaseBinding.java)

I noticed this PR has EventGridTrigger is missing them.

Since annotation type definitions can't implement explicitly, I'm proposing that at least we get a unit test in place to verify that `name()` and `dataType()' are part of any annotation under `functions` package, particularly since we're building up a lot of efferent coupling in Functions.

File: azure-functions-java-core/src/main/java/com/microsoft/azure/serverless/functions/annotation/EventGridTrigger.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+package com.microsoft.azure.serverless.functions.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.PARAMETER, ElementType.METHOD})
```
Comment (xscript) [inline azure-functions-java-core/src/main/java/com/microsoft/azure/serverless/functions/annotation/EventGridTrigger.java]: Triggers should be only applied to `ElementType.Parameter`.

File: azure-functions-java-core/src/main/java/com/microsoft/azure/serverless/functions/annotation/EventGridTrigger.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+public @interface EventGridTrigger {
+
```
Comment (xscript) [inline azure-functions-java-core/src/main/java/com/microsoft/azure/serverless/functions/annotation/EventGridTrigger.java]: As @kevinhillinger pointed out, `name` and `dataType` are required fields.

**Comment by xscript** [comment]
@kevinhillinger You are right. Ideally, we should put common fields to a base class/annotation. But Java doesn't allow annotation inheritance. So we have to put them in every single annotation.

As to you proposal, we also have some non-binding annotations:
-  [FunctionName](https://github.com/Azure/azure-functions-java-worker/blob/dev/azure-functions-java-core/src/main/java/com/microsoft/azure/serverless/functions/annotation/FunctionName.java)
- [StorageAccount](https://github.com/Azure/azure-functions-java-worker/blob/dev/azure-functions-java-core/src/main/java/com/microsoft/azure/serverless/functions/annotation/StorageAccount.javal)
-  [BindingName](https://github.com/Azure/azure-functions-java-worker/blob/dev/azure-functions-java-core/src/main/java/com/microsoft/azure/serverless/functions/annotation/BindingName.java)

You should put them into a white-list when adding unit tests for annotations.

**Comment by kevinhillinger** [comment]
@christopheranderson can these changes be incorporated into the PR? There's situations coming up for being able to use Event Grid trigger, and this will take it across the finish line.

**Pull request closed without merging**